lib/_http_common.js

```diff
@@ -217,15 +217,20 @@ function checkIsHttpToken(val) {
   return tokenRegExp.test(val);
 }
 
-const headerCharRegex = /[^\t\x20-\x7e\x80-\xff]/;
+const headerCharRegex = /([^\t\x20-\x7e\x80-\xff])/;
 /**
- * True if val contains an invalid field-vchar
+ * Returns object if val contains an invalid field-vchar
  *  field-value    = *( field-content / obs-fold )
  *  field-content  = field-vchar [ 1*( SP / HTAB ) field-vchar ]
  *  field-vchar    = VCHAR / obs-text
  */
 function checkInvalidHeaderChar(val) {
-  return headerCharRegex.test(val);
+  if (!headerCharRegex.test(val)) return null;
```

Comment:
nit: Returning `undefined` here (explicitly or implicitly) should be just fine.

```suggestion
  if (!headerCharRegex.test(val)) return undefined;
```

Reply:
I don't know if it matters (optimizer-wise) to modern versions of V8 or not, but at least by returning `null`, this function will technically always be returning a type of `'object'`.